src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -327,4 +323,24 @@ void SoundRecorder::cleanup()
     captureDevice = nullptr;
 }
 
+
+////////////////////////////////////////////////////////////
+void SoundRecorder::launchCapturingThread()
+{
+    m_isCapturing = true;
+
+    assert(!m_thread.joinable());
+    m_thread = std::thread(&SoundRecorder::record, this);
+}
+
+
+////////////////////////////////////////////////////////////
+void SoundRecorder::awaitCapturingThread()
+{
+    m_isCapturing = false;
+
+    if (m_thread.joinable())
```

Comment:
I'm not quite following why we "allow" non-joinable threads...
Are the certain platforms that don't allow for it?

Reply:
`std::thread::joinable()` returns whether the thread is currently active or not (suitable to be joined), not whether the platform is capable or joining or not .Calling `std::thread::join()` requires `std::thread::joinable()` to return `true` as a precondition, otherwise an exception will be thrown. 

`sf::Thread::wait()` does not have such a precondition, and looking at `ThreadImpl.cpp` checking if the thread is active performed internally inside `ThreadImpl::wait()`. 

I am explicitly checking for `std::thread::joinable()` here as I am not sure what the state of the thread is, and I want to maintain the exact same semantics as we had before. If I were to remove the joinable check, the semantics would differ from what `sf::Thread::wait()` used to do. Maybe the check is not necessary at all, but that is outside of the scope of this PR, as I just want to replace SFML-specific threading primitives with standard ones, avoiding or minimizing any semantical change to the behavior of SFML.